internal/jobcontainers/process.go

```diff
@@ -0,0 +1,233 @@
+package jobcontainers
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"sync"
+
+	"github.com/Microsoft/hcsshim/internal/cow"
+	"github.com/Microsoft/hcsshim/internal/guestrequest"
+	"github.com/Microsoft/hcsshim/internal/hcs"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+// JobProcess represents a process run in a job object.
+type JobProcess struct {
+	cmd            *exec.Cmd
+	procLock       sync.Mutex
+	stdioLock      sync.Mutex
+	stdin          io.WriteCloser
+	stdout         io.ReadCloser
+	stderr         io.ReadCloser
+	waitBlock      chan struct{}
+	closedWaitOnce sync.Once
+	waitError      error
+}
+
+var sigMap = map[string]int{
+	"CtrlC":        windows.CTRL_BREAK_EVENT,
+	"CtrlBreak":    windows.CTRL_BREAK_EVENT,
+	"CtrlClose":    windows.CTRL_CLOSE_EVENT,
+	"CtrlLogOff":   windows.CTRL_LOGOFF_EVENT,
+	"CtrlShutdown": windows.CTRL_SHUTDOWN_EVENT,
+}
+
+var _ cow.Process = &JobProcess{}
+
+func newProcess(cmd *exec.Cmd) *JobProcess {
+	return &JobProcess{
+		cmd:       cmd,
+		waitBlock: make(chan struct{}),
+	}
+}
+
+func (p *JobProcess) ResizeConsole(ctx context.Context, width, height uint16) error {
+	return nil
```

Comment:
Do we have future work planned to support ConPTY and `EmulateConsole`?

Reply:
Yep, the handle (I think, it's been awhile) needs to be specified on the thread attributes for the job just like we're planning to do for starting the process in the job right when it's launched so those will come together. I got ConPTY to work but there was some oddities with the binding, even though the definitions are beyond simple, where it would return invalid parameter or something on process startup. We're best friends with the console folks now (😄) so I'll talk to them if I run into it again

Reply:
Should we block running with `EmulateConsole = true` until that is in place?

Reply:
Yea probably a good bet. EmulateConsole just gets set if Terminal is true on the OCI spec, so just block if this is set?

Reply:
Done